python/tvm/meta_schedule/runner/local_runner.py

```diff
@@ -0,0 +1,341 @@
+        f_cleanup: Optional[str] = None,
+        initializer: Optional[Callable[[], None]] = None,
+    ) -> None:
+        super().__init__()
+        self.timeout_sec = timeout_sec
+        self.evaluator_config = EvaluatorConfig._normalized(evaluator_config)
+        self.cooldown_sec = cooldown_sec
+        self.alloc_repeat = alloc_repeat
+        self.f_alloc_argument = f_alloc_argument
+        self.f_run_evaluator = f_run_evaluator
+        self.f_cleanup = f_cleanup
+
+        self.pool = PopenPoolExecutor(
+            max_workers=1,  # one local worker
+            timeout=timeout_sec,
+            initializer=initializer,
+        )
+
+    def run(self, runner_inputs: List[RunnerInput]) -> List[RunnerFuture]:
+        results: List[RunnerFuture] = []
+        for runner_input in runner_inputs:
+            future = self.pool.submit(
+                LocalRunner._worker_func,
+                self.f_alloc_argument,
+                self.f_run_evaluator,
+                self.f_cleanup,
+                self.evaluator_config,
+                self.alloc_repeat,
+                str(runner_input.artifact_path),
+                str(runner_input.device_type),
+                tuple(arg_info.as_json() for arg_info in runner_input.args_info),
+            )
+            try:
+                result: List[float] = future.result()
+                error_message: str = None
+            except TimeoutError as exception:
+                result: List[float] = None
+                error_message: str = (
+                    f"LocalRunner: Timeout, killed after {self.timeout_sec} seconds\n"
+                )
+            except Exception as exception:  # pylint: disable=broad-except
+                result: List[float] = None
+                error_message: str = "LocalRunner: An exception occurred\n" + str(exception)
+            local_future = LocalRunnerFuture(res=result, error_message=error_message)
+            results.append(local_future)
+        return results
+
+    @staticmethod
+    def _worker_func(
+        _f_alloc_argument: Optional[str],
+        _f_run_evaluator: Optional[str],
+        _f_cleanup: Optional[str],
+        evaluator_config: EvaluatorConfig,
+        alloc_repeat: int,
+        artifact_path: str,
+        device_type: str,
+        args_info: T_ARG_INFO_JSON_OBJ_LIST,
+    ) -> List[float]:
+        f_alloc_argument: LocalRunner.T_ALLOC_ARGUMENT = get_global_func_with_default_on_worker(
+            _f_alloc_argument, default_alloc_argument
+        )
+        f_run_evaluator: LocalRunner.T_RUN_EVALUATOR = get_global_func_with_default_on_worker(
+            _f_run_evaluator, default_run_evaluator
+        )
+        f_cleanup: LocalRunner.T_CLEANUP = get_global_func_with_default_on_worker(
+            _f_cleanup, default_cleanup
+        )
+
+        @contextmanager
+        def resource_handler():
+            try:
+                yield
+            finally:
+                # Final step. Always clean up…
```

Comment:
Let's stick with ascii for now. I remember once windows build was broken because of non-ascii code

```suggestion
                # Final step. Always clean up
```

Reply:
done